decentraland/unity-explorer
# featurette: World Interaction SFX + avatar fixes

## What does this PR change?

_**->It changes your life duuudeee ->**_

Also, it adds SFX when interacting with world elements that accept clicks, E and F inputs.
For this I created a new system that runs before WritePointerEventResultsSystem (because that system deletes stuff that we need after writing it to the CRDT)

I have also added here the fix for the avatar long fall sound, now its a sound in loop that stops once the player reaches the ground. The sound itself might need changing, I could not find in the old renderer which clip was played when playing, so for now its just "clothes rustling" sound.
I have added some functionality so the sound does not repeat itself but plays another random clip from the clips loaded on the config for that sound and made that method static in the utils for audio so it can be reused later for other SFX that might need the same logic.

I also added here some missing sounds for the standard emotes, which will also help reduce a bit some errors that appear when doing them related to missing animation events.

Also, in a tech side, I removed the AudioPlaybackContainer and added to it the LandscapeAudioSystem so now its all in the AudioPlaybackPlugin. After Vitaly changes I could do it in a very straightforward manner as the TerrainGenerator was populated earlier on the LandscapeSystem.


## How to test the changes?

To test, just go to any interactable SDK element and interact with them. They should highlight and the mouse change to a pointer and ask for input (pointer, E or F should work).

For the avatar long falling sound, just fall from a great height xD

And for the emotes, just play the basic emotes, they should have sounds now (not the stickers with hearts or snowflakes thou, that's a whole another can of worms I did not open) and there should not be any errors triggered by missing animation events.
